connectors/spark/src/test/java/io/unitycatalog/spark/ApiClientFactoryTest.java

```diff
@@ -0,0 +1,128 @@
+package io.unitycatalog.spark;
+
+import io.unitycatalog.client.ApiClient;
+import org.junit.jupiter.api.Test;
+
+import java.net.URI;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Test class for ApiClientFactory to verify User-Agent configuration.
+ */
+public class ApiClientFactoryTest {
+
+  @Test
+  public void testUserAgentContainsSparkVersion() throws Exception {
+    URI uri = new URI("http://localhost:8080");
+    ApiClient client = ApiClientFactory.createApiClient(uri, null);
+
+    String userAgent = client.getUserAgent();
+    System.out.println("User-Agent: " + userAgent);
+
+    // Verify the user agent contains the base Unity Catalog client
+    assertThat(userAgent).startsWith("UnityCatalog-Java-Client/");
+
+    // Verify the user agent contains Spark version
+    assertThat(userAgent).contains("Spark/");
+
+    // Extract and verify Spark version is not empty
+    String[] parts = userAgent.split(" ");
+    boolean foundSpark = false;
+    for (String part : parts) {
+      if (part.startsWith("Spark/")) {
+        String version = part.substring("Spark/".length());
+        assertThat(version).isNotEmpty();
+        foundSpark = true;
+        System.out.println("Found Spark version: " + version);
+        break;
+      }
+    }
+    assertThat(foundSpark).isTrue();
+  }
+
+  @Test
+  public void testUserAgentMayContainDeltaVersion() throws Exception {
+    URI uri = new URI("http://localhost:8080");
+    ApiClient client = ApiClientFactory.createApiClient(uri, null);
+
+    String userAgent = client.getUserAgent();
+    System.out.println("User-Agent: " + userAgent);
+
+    // Delta version is optional (depends on classpath)
+    // If Delta is available, verify the version is present
+    if (userAgent.contains("Delta/")) {
```

Comment:
verify the Delta must be present, even if the version is optional. throw error if not Delta not present